File: swift.swift
#!/usr/bin/env swift
import Foundation

var process = Process()
process.executableURL = URL(fileURLWithPath: "/usr/bin/env")
process.arguments = ["swift"] + ProcessInfo.processInfo.arguments

let standardOutputPipe = Pipe()
standardOutputPipe.fileHandleForReading.readabilityHandler = { fileHandle in
    let data = fileHandle.availableData

    try! FileHandle.standardOutput.write(contentsOf: data)
}
process.standardOutput = standardOutputPipe

let standardErrorPipe = Pipe()
standardErrorPipe.fileHandleForReading.readabilityHandler = { fileHandle in
    let data = fileHandle.availableData

    try! FileHandle.standardError.write(contentsOf: data)
}
process.standardError = standardErrorPipe

try! process.run()
process.waitUntilExit()

File: swift.py
#!/usr/bin/env python
import sys
import subprocess

swift = subprocess.Popen(["swift", *sys.argv], stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True)

while swift.poll() is None:
    stdout = swift.stdout.readline()
    stderr = swift.stderr.readline()

    print(stdout, end=None, file=sys.stdout)
    print(stderr, end=None, file=sys.stderr)
